core/trino-main/src/main/java/io/trino/operator/scalar/JsonStringArrayExtractScalar.java

```diff
@@ -0,0 +1,159 @@
+        ArrayBlockBuilderAppender arrayAppender = new ArrayBlockBuilderAppender(new VarcharBlockBuilderAppender(arrayType.getElementType()));
+        MethodHandle methodHandle = METHOD_HANDLE.bindTo(arrayType).bindTo(arrayAppender);
+
+        return new ChoicesSpecializedSqlScalarFunction(
+                boundSignature,
+                NULLABLE_RETURN,
+                ImmutableList.of(NEVER_NULL, NEVER_NULL),
+                methodHandle);
+    }
+
+    @UsedByGeneratedCode
+    public static Block extract(ArrayType arrayType, ArrayBlockBuilderAppender arrayAppender, ConnectorSession connectorSession, Slice json, JsonPath jsonPath)
+    {
+        try (JsonParser jsonParser = createJsonParser(JSON_FACTORY, json)) {
+            jsonParser.nextToken();
+            if (jsonParser.getCurrentToken() == JsonToken.VALUE_NULL) {
+                return null;
+            }
+            BlockBuilder blockBuilder = arrayType.createBlockBuilder(null, 1);
+            arrayAppender.append(jsonParser, jsonPath, blockBuilder);
+            Block block = blockBuilder.build();
+            return arrayType.getObject(block, 0);
+        }
+        catch (TrinoException | JsonCastException e) {
+            throw new TrinoException(INVALID_CAST_ARGUMENT, format("Cannot cast to %s. %s\n%s", arrayType, e.getMessage(), truncateIfNecessaryForErrorMessage(json)), e);
+        }
+        catch (Exception e) {
+            throw new TrinoException(INVALID_CAST_ARGUMENT, format("Cannot cast to %s.\n%s", arrayType, truncateIfNecessaryForErrorMessage(json)), e);
+        }
+    }
+
+    record VarcharBlockBuilderAppender(Type type)
+    {
+        public void append(Slice slice, BlockBuilder blockBuilder)
+        {
+            if (slice == null) {
+                blockBuilder.appendNull();
+            }
+            else {
+                type.writeSlice(blockBuilder, slice);
+            }
+        }
+    }
+
+    public record ArrayBlockBuilderAppender(VarcharBlockBuilderAppender elementAppender)
+    {
+        public void append(JsonParser parser, JsonPath path, BlockBuilder blockBuilder)
+                throws IOException
+        {
+            if (parser.getCurrentToken() == JsonToken.VALUE_NULL) {
+                blockBuilder.appendNull();
+                return;
+            }
+
+            if (parser.getCurrentToken() != START_ARRAY) {
+                throw new JsonCastException(format("Expected a json array, but got %s", parser.getText()));
+            }
+            ((ArrayBlockBuilder) blockBuilder).buildEntry(elementBuilder -> {
+                while (parser.nextToken() != END_ARRAY) {
+                    if (parser.getCurrentToken() == JsonToken.VALUE_NULL) {
+                        elementAppender.append(null, elementBuilder);
+                        continue;
+                    }
+                    JsonParser jsonParser = parser.readValueAsTree().traverse();
+                    jsonParser.nextToken();
```

Comment:
shouldn't you check token is not null?

Reply:
all requierd checks are within extractor.